### PR TITLE
Some KafkaSource scaling rekt tests.

### PR DESCRIPTION
**Pull request by lionelvillard**
## Proposed Changes

- Add one scaling test sending event for 20s while scaling up the KafkaSource to 200 consumers. Check if there is no events loss.
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: test/test_images/kafka-proxy/main.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
+/*
```
Comment (aliok) [inline test/test_images/kafka-proxy/main.go:1]: Can you add a README file within this folder for this test image? What it does, how to configure it, etc.

Only very briefly as it is not a deliverable of the Knative project

Comment (lionelvillard) [reply test/test_images/kafka-proxy/main.go:1]: good idea! Will do.

File: test/rekt/features/kafkasource/scaling.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	// Setup:
+	// Sender (eventshub) -> KafkaProxy -> KafkaTopic -> KafkaSource -> Sink (eventshub)
+	proxyName := feature.MakeRandomK8sName("proxy")
+	f.Setup("set proxy name state", func(ctx context.Context, t feature.T) {
+		state.SetOrFail(ctx, t, "proxyName", proxyName)
+	})
+
+	f.Setup("install Kafka proxy", kafkaproxy.Install(proxyName,
+		kafkaproxy.WithBootstrapServer(kafkaBootstrapUrlPlain),
+		kafkaproxy.WithTopic(topic)))
```
[the remark this answers is not in the document]
Comment (lionelvillard) [reply test/rekt/features/kafkasource/scaling.go:54]: it's done here: https://github.com/knative-sandbox/eventing-kafka/blob/c20df2dbbcfc35242f50dd2069fea3ca3f5a63cf/test/rekt/resources/kafkaproxy/kafkaproxy.go#L54

However it does not wait for the endpoint to be ready, only the pod, which is wrong. I will fix it. Thanks for pointing me to `k8s.WaitForServiceEndpointsOrFail`!

File: test/rekt/features/kafkasource/scaling.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	)(ctx, t)
+
+	time.Sleep(2 * time.Second)
+
+	env := environment.FromContext(ctx)
+	source, err := kafkaclient.Get(ctx).SourcesV1beta1().KafkaSources(env.Namespace()).Get(ctx, sourceName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get the KafkaSource object: %v", err)
+	}
+	source.Spec.Consumers = pointer.Int32Ptr(130)
```
Comment (aliok) [inline test/rekt/features/kafkasource/scaling.go:128]: Just a random number?

**Comment by aliok** [comment]
@lionelvillard this test is to make sure the problem https://github.com/knative-sandbox/eventing-kafka/issues/411 is fixed? What's the relation between them?

File: test/rekt/features/kafkasource/scaling.go (hunk 3 above)
Comment (lionelvillard) [reply test/rekt/features/kafkasource/scaling.go:128]: not totally. It has to be over 100 because each pod can have 100 "virtual replicas".

**Comment by lionelvillard** [comment]
> @lionelvillard this test is to make sure the problem #411 is fixed? What's the relation between them?

Yup

**Comment by lionelvillard** [comment]
no more rekt. Will do something else

**Pull request closed without merging**